File: Methods-Java/Java source-code/variance.java


public class variance {
    public static double variance_m(Double[] x) {
        double sum = 0;
        double sum1 = 0;
        double var = 0;
        double avrg = 0;
        for (int i = 0; i < x.length; i++) {
            sum = sum + x[i];
        }
        avrg = sum / (double) x.length;
        for (int i = 0; i < x.length; i++) {
            sum1 = sum1 + (x[i] - avrg) * (x[i] - avrg);
        }
        var = sum1 / (double) x.length;
        return var;
    }
}

File: Methods/Methods-Python/variance.py
def variance(x):
    suma = 0
    sum1 = 0
    var = 0
    avrg = 0

    for i in range(0, len(x)):
        suma = suma + x[i]

    avrg = suma / len(x)

    for i in range(0, len(x)):
        sum1 = sum1 + (x[i] - avrg) * (x[i] - avrg)

    var = sum1 / len(x)

    return var
